core/loader_shared.c

```diff
@@ -112,6 +112,9 @@ loader_init(void)
         char name_copy[MAXIMUM_PATH];
         mod = privload_insert(NULL, privmod_static[i].base, privmod_static[i].size,
                               privmod_static[i].name, privmod_static[i].path);
+#ifdef CLIENT_INTERFACE
+        mod->is_client = true;
```

Comment:
Are you sure these are all client libs?  This seems fragile: if we later load sthg else early this will break.  Seems best to have some kind of sanity check: at least add an assert in the current location where this is set.

Reply:
Yes, to my understanding this is always a client lib. It's the client libraries that preliminarily get loaded before the heap is set up, w/o getting finalized and their imports loaded. App dlls are not part of this. The only place I had looked at to assert this is at loader.c:573:

```
/* i#852: identify all libs that import from DR as client libs */
   if (impmod->base == get_dynamorio_dll_start())
        mod->is_client = true;
```

And this code is stale. This is never hit because libdynamorio.so at this point is loaded already so privload_lookup("libdynamorio.so") is never NULL. Should also investigate this, I'll add a XXX. WDYT?

Reply:
That is where I was thinking the assert would go: assert !mod->is_client on an else of that.  Hmm, you're saying it's never hit: so mod->is_client is never set to true?

Reply:
Just ran an experiment. Yes. Was libdynamorio.so loaded differently or at a different point in time in the past?

Reply:
Not sure: maybe it wasn't added to the list until later.

Reply:
I've added a comment.